### PR TITLE
fix cursor image on hover when there're multiple creatures nearby

### LINKED ISSUE #800: cursor shows "pick object" instead of "attack"
My steps were:

1. enter monolith, hero has no more turns, plan to attack Basilisks
2. end turn
3. hover Basilisks - cursor shows "pick object" instead of "attack"

Screenshot 2022-09-03 at 11 48 06
Screenshot 2022-09-03 at 11 48 28
Screenshot 2022-09-03 at 11 48 54

### PR DESCRIPTION
**Pull request by kambala-decapitator**
fixes #800, you can find investigation details there

### CODE REVIEW

File: client/windows/CAdvmapInterface.cpp [hunk 1]
```diff
@@ -1711,8 +1711,7 @@ void CAdvMapInt::tileHovered(const int3 &mapPos)
 	}
 	else if(const CGHeroInstance * h = curHero())
 	{
-		int3 mapPosCopy = mapPos;
-		const CGPathNode * pnode = LOCPLINT->cb->getPathsInfo(h)->getPathInfo(mapPosCopy);
+		const CGPathNode * pnode = LOCPLINT->cb->getPathsInfo(h)->getPathInfo(mapPos);
```
Comment (Nordsoft91) [inline client/windows/CAdvmapInterface.cpp:1714]: does getPathInfo receives const int3 &?

File: lib/CPathfinder.cpp [hunk 2]
```diff
@@ -862,7 +862,7 @@ void DestinationActionRule::process(
 				else
 					action = CGPathNode::BLOCKING_VISIT;
 			}
-			else if(destination.isGuardianTile)
+			else if(destination.guarded)
```
Comment (Nordsoft91) [inline lib/CPathfinder.cpp:865]: guarded will return true for neighbour tiles too?

File: client/windows/CAdvmapInterface.cpp (hunk 1 above)
Comment (kambala-decapitator) [reply client/windows/CAdvmapInterface.cpp:1714]: yes, otherwise it wouldn't have compiled

File: lib/CPathfinder.cpp (hunk 2 above)
Comment (kambala-decapitator) [reply lib/CPathfinder.cpp:865]: yes, as far as I understood. You can also see tested cases in #800

**Review by Nordsoft91 — APPROVED** [review]

**Pull request merged**